metrics/cli/test/deploy/deployer_test.dart

```diff
@@ -0,0 +1,395 @@
+
+    test(
+      "throws an ArgumentError if the given Firebase command is null",
+      () {
+        expect(
+          () => Deployer(
+            services: services,
+            firebaseCommand: null,
+            gitCommand: gitCommand,
+            fileHelper: fileHelper,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given git command is null",
+      () {
+        expect(
+          () => Deployer(
+            services: services,
+            firebaseCommand: firebaseCommand,
+            gitCommand: null,
+            fileHelper: fileHelper,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given file helper is null",
+      () {
+        expect(
+          () => Deployer(
+            services: services,
+            firebaseCommand: firebaseCommand,
+            gitCommand: gitCommand,
+            fileHelper: null,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".deploy() logs in to the GCloud",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verify(gcloudService.login()).called(once);
+      },
+    );
+
+    test(
+      ".deploy() logs in to the GCloud before creating the GCloud project",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verifyInOrder([
+          gcloudService.login(),
+          gcloudService.createProject(),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() creates the GCloud project",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verify(gcloudService.createProject()).called(once);
+      },
+    );
+
+    test(
+      ".deploy() creates the GCloud project before login to the Firebase",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verifyInOrder([
+          gcloudService.createProject(),
+          firebaseCommand.login(),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() logs in to the Firebase",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verify(gcloudService.createProject()).called(once);
+      },
+    );
+
+    test(
+      ".deploy() logs in to the Firebase before adding the Firebase capabilities to the project",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        await deployer.deploy();
+
+        verifyInOrder([
+          firebaseCommand.login(),
+          firebaseCommand.addFirebase(any, any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() adds the Firebase capabilities to the created project",
+      () async {
+        const projectId = 'testId';
+        const firebaseToken = 'testToken';
```

Comment:
Would it be great to make these constants common for all tests to avoid duplication?